lib/src/lib/button/directives/preventDoubleClick.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { Directive, EventEmitter, HostListener, Output } from '@angular/core';
+import { Subject, timer } from 'rxjs';
+import { throttle } from 'rxjs/operators';
+
+@Directive({
+  selector: '[lab900PreventDoubleClick]',
+  standalone: true,
+})
+export class PreventDoubleClickDirective {
+  private clicks = new Subject<Event>();
+  private isThrottled = false;
+
+  @Output() public throttledClick = new EventEmitter<Event>();
+
+  public constructor() {
+    this.clicks.pipe(throttle(() => timer(1000))).subscribe((event) => {
```

Comment:
don't forget to cleanup your subscription.
-> add a `takeUntilDestroyed()` at the end of the pipe